tests/backup/backup_test.go

```diff
@@ -6850,8 +6851,56 @@ var _ = Describe("{SwapShareBackup}", func() {
 			err := Inst().S.Destroy(contexts[i], opts)
 			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
 		}
-		var wg sync.WaitGroup
 
+		log.InfoD("Deleting all restores")
+		for _, userName := range users {
+			ctx, err := backup.GetNonAdminCtx(userName, "Password1")
+			log.FailOnError(err, "Fetching nonAdminctx")
+			allRestores, err := GetAllRestoresNonAdminCtx(ctx)
+			log.FailOnError(err, "Fetching all restors for nonAdminctx")
+			for _, restoreName := range allRestores {
+				err = DeleteRestore(restoreName, orgID, ctx)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Verifying restore deletion - %s", restoreName))
+			}
+		}
+
+		log.InfoD("Delete all backups")
+		for i := 0; i <= numberOfUsers-1; i++ {
+			ctx, err := backup.GetNonAdminCtx(users[i], "Password1")
+			log.FailOnError(err, "Fetching nonAdminctx ")
+			_, err = DeleteBackup(backupName, backupUIDList[i], orgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verifying backup deletion - %s", backupName))
+		}
+
+		// Cleanup all backup locations
+		for _, userName := range users {
+			ctx, err := backup.GetNonAdminCtx(userName, "Password1")
+			log.FailOnError(err, "Fetching nonAdminctx ")
+			allBackupLocations, err := getAllBackupLocations(ctx)
+			dash.VerifySafely(err, nil, "Verifying fetching of all backup locations")
+			for backupLocationUid, backupLocationName := range allBackupLocations {
+				if userBackupLocationMapping[userName] == backupLocationName {
+					err = DeleteBackupLocation(backupLocationName, backupLocationUid, orgID)
+					dash.VerifySafely(err, nil, fmt.Sprintf("Verifying backup location deletion - %s", backupLocationName))
+				}
+			}
+
+			backupLocationDeletionSuccess := func() (interface{}, bool, error) {
+				allBackupLocations, err := getAllBackupLocations(ctx)
+				dash.VerifySafely(err, nil, "Verifying fetching of all backup locations")
+				for _, backupLocationName := range allBackupLocations {
+					if userBackupLocationMapping[userName] == backupLocationName {
+						return "", true, fmt.Errorf("found %s backup locations", backupLocationName)
+					} else {
```

Comment:
can you please check this logic. i  think this might fail when there are multiple backup location for one user

Reply:
resolved